unified_planning/engines/factory.py

```diff
@@ -531,8 +531,14 @@ def _get_engine_class(
                 )
                 planners_features.append(x)
         if len(planners_features) > 0:
+            if optimality_guarantee is not None:
+                starting_line = f"No available engine supports all the problem features with optimality_guarantee: {optimality_guarantee.name}:"
+            elif anytime_guarantee is not None:
+                starting_line = f"No available engine supports all the problem features with anytime_guarantee: {anytime_guarantee.name}:"
+            else:
+                starting_line = "No available engine supports all the problem features:"
             header = ["Engine"] + problem_features
-            msg = f"No available engine supports all the problem features:\n{format_table(header, planners_features)}"
```

Comment:
Here we should add a column to indicate for each engine if it supports the optimality guarantee